storage_service/locations/models/package.py

```
@@ -2602,20 +2506,16 @@ def finish_reingest(
                         " PREMIS events".format(compression)
                     )
                 except KeyError:
-                    LOGGER.error(
-                        "Failed to extract valid compression algorithm from"
-                        ' "{}"; does not match any of the following recognized'
-                        " options: {}".format(
-                            compression_algorithm, ", ".join(utils.COMPRESSION_ALGORITHMS)
-                        )
-                    )
-                    raise StorageException(
+                    msg = (
                         "Failed to extract valid compression algorithm from"
                         ' "{}"; does not match any of the following recognized'
                         " options: {}".format(
-                            compression_algorithm, ", ".join(utils.COMPRESSION_ALGORITHMS)
+                            compression_algorithm,
+                            ", ".join(utils.COMPRESSION_ALGORITHMS),
                         )
                     )
+                    LOGGER.error(msg)
+                    raise StorageException(msg)
```

Perhaps remove the logging in favor of `raise StorageException(msg)` my understanding is both should return to the console? What do you think?

---

I think i put this line in because it seemed that the error was just being swallowed (i.e. I was getting 500 errors on the API call and nothing logged) - this being called by `obj_update_hook` on the tastypie endpoint. It's possible there's something in my setup that's causing this but I'm not sure.

---

@helenst I have been having this exact issue today trying to debug S3 so I'd keep it if you haven't removed it already. I hadn't quire realized how little the storage service was logging when things weren't quite performing as expected.